### DISCUSSION TITLE
pnpm allowed pre post scripts

### DISCUSSION THREAD

**Discussion by TorbjornHoltmon** [pnpm/pnpm#5462] [Ideas]
We can allow pre and post scripts with `enable-pre-post-scripts=true`.

This allows all pre post scripts.

I would like to enable only the post scripts I explicitly allow.

Something like:
```json
  "pnpm": {
    "allowedPrePostScripts":  ["self", "husky", "some-package"]
  }
```

Additionally, an interactive install/add option to prompt the user if they want to add pre/post script to the allowed list.

`pnpm install --interactive-pre-post-scripts`

pnpm will then see what packages have pre and post scripts, and prompt the user if they want to run them and add them to the allowed list.

**Comment by zkochan** [comment]
This is a lot of functionality for a feature that would be almost never used.
